my-app/src/Button/ButtonClass.tsx

```diff
@@ -1,29 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import styles from './button.module.css';
 
 type Props = {
     visible: boolean;
-    // handleClick: () => void;
+    handleClick: () => void;
 }
 
-type State = {
-    opened: boolean
-}
-export class ButtonComponent extends React.Component<Props, State> {
-    state = {opened: !!this.props.visible}
-    
-    click = () => {
-        // this.props.handleClick()
-        this.setState({opened: !this.state.opened})
-    }
+export const Button = ({ visible, handleClick }: Props) => {
+    const [isVisible, setIsVisible] = useState(visible)
 
-    render () {
-        return (
-            <button className={styles.button} onClick={this.click}>{this.state.opened ? <Burger /> : <Close />}  </button>
-        )
+    const click = () => {
+        handleClick()
+        setIsVisible(!isVisible)
     }
 
+    return (
+        <button className={styles.button} onClick={click}>{visible ? <Burger /> : <Close />}  </button>
```

Comment:
you check prop visible, that stay same all time
you need to check state

Reply:
i fixed it